cluster-autoscaler/core/utils.go

```diff
@@ -106,17 +108,33 @@ func (podMap podSchedulableMap) set(pod *apiv1.Pod, err *simulator.PredicateErro
 	})
 }
 
-// FilterOutSchedulable checks whether pods from <unschedulableCandidates> marked as unschedulable
+// filterOutSchedulable checks whether pods from <unschedulableCandidates> marked as unschedulable
 // by Scheduler actually can't be scheduled on any node and filter out the ones that can.
 // It takes into account pods that are bound to node and will be scheduled after lower priority pod preemption.
-func FilterOutSchedulable(unschedulableCandidates []*apiv1.Pod, nodes []*apiv1.Node, allScheduled []*apiv1.Pod, podsWaitingForLowerPriorityPreemption []*apiv1.Pod,
-	predicateChecker *simulator.PredicateChecker, expendablePodsPriorityCutoff int) []*apiv1.Pod {
-	unschedulablePods := []*apiv1.Pod{}
+// TODO: filterOutSchedulable does too much to be a util func. Should be made a unexported method of StaticAutoScaler.
+// TODO: Also has too many parameters
+func filterOutSchedulable(unschedulableCandidates []*apiv1.Pod, nodes []*apiv1.Node, allScheduled []*apiv1.Pod, podsWaitingForLowerPriorityPreemption []*apiv1.Pod,
+	predicateChecker *simulator.PredicateChecker, expendablePodsPriorityCutoff int, excludeSchedulablePods bool) []*apiv1.Pod {
+	unschedulableCandidatesWithNominatedNode, unschedulableCandidates := splitPodsWithNominatedNode(unschedulableCandidates)
+	if !excludeSchedulablePods {
+		return unschedulableCandidates
+	}
+
+	var unschedulablePods []*apiv1.Pod
 	nonExpendableScheduled := FilterOutExpendablePods(allScheduled, expendablePodsPriorityCutoff)
 	nodeNameToNodeInfo := scheduler_util.CreateNodeNameToInfoMap(append(nonExpendableScheduled, podsWaitingForLowerPriorityPreemption...), nodes)
 	podSchedulable := make(podSchedulableMap)
 	loggingQuota := glogx.PodsLoggingQuota()
 
+	for _, pod := range unschedulableCandidatesWithNominatedNode {
+		nominatedNode := pod.Status.NominatedNodeName
+		nodeNameToNodeInfo[nominatedNode] = scheduler_util.NodeWithPod(nodeNameToNodeInfo[nominatedNode], pod)
+	}
+
+	sort.Slice(unschedulableCandidates, func(i, j int) bool {
```

Comment:
I would want to change this sort function to still sort by Pod size when priority is same. This would correct a failing test too (https://github.com/kubernetes/autoscaler/pull/1596). I'm not sure if this is the right thing to do as I'm speculating on the behaviour of the scheduler

Reply:
idk. I suggest to leave it. It would add some complexity - yet I am not sure if any benefit. 
As you pointed out it would not simulate actual scheduler behavior.

Reply:
Sure. Thanks!